server/src/main/java/org/opensearch/cluster/coordination/Coordinator.java

```diff
@@ -181,6 +183,8 @@ public class Coordinator extends AbstractLifecycleComponent implements Discovery
     private JoinHelper.JoinAccumulator joinAccumulator;
     private Optional<CoordinatorPublication> currentPublication = Optional.empty();
     private final NodeHealthService nodeHealthService;
+    private final RemoteClusterStateService remoteClusterStateService;
```

Comment:
do we need to take a dependency on RemoteClusterStateService here? from what i understand we just send updates to RemotePersistedState and it is responsible for persisting in remote via RemoteClusterStateService?